### ISSUE TITLE
Add defaults for headers, e.g. content-security-policy

### ISSUE THREAD

**Issue by paulgibbs** [humanmade/hm-platform#118]
I always forget to set certain headers on client sites relating to browser security.
https://github.com/humanmade/Human-Made-Website-Theme/issues/497 has some examples of a few headers that might be a minimum baseline.

What do you think of adding a little plugin that loads inside HM Platform that sets these, so the minimums are always in place, while allowing client sites to upgrade from them?

**Comment by nathanielks** [comment]
I'm in favor 👍 @joehoyle? We could also potentially add these at the infrastructure level.

**Comment by Japh** [comment]
This isn't really infrastructure, and everything we run should have `hm-platform`, so I think this is the right place for it.

Thanks for the suggestion, Paul!

**Comment by nathanielks** [comment]
Semantic? Would you clarify that?

**Comment by Japh** [comment]
To use the example of `content-security-policy`, which seems to me semantically related to the application that's managing and serving the content, rather than the infrastructure which is supporting the application.

**Comment by joehoyle** [comment]
Is it possible someone _wouldn't_ want these headers? That's probably reason for it to exist in hm-platform as it's probably easier to override.

**Comment by nathanielks** [comment]
Yeah, that makes sense 👍  If it was something we wanted on all the time, no matter what, I'd argue let's bake this into our nginx config. If it's configurable, let's not!

**Comment by joehoyle** [comment]
This was raised in https://github.com/humanmade/hosting-support/issues/426, I'm not super familiar, but how much crafting does a CSP need? It's my understand it's quite dependant on the website.
